### PR TITLE
fix: Add translators for Media and Unknown Events

### PR DESCRIPTION
**Pull request by alexs-mparticle**
## Summary
- When generating snippets for iOS and Web, Media Events are appearing as "Other" and Unknown Events appear as "Media".

## Testing Plan
- Build and run the Demo project. Use the following Data plan and confirm that Media and Unknown Events appear properly on all platforms.

```json
{
  "version": 1,
  "data_plan_id": "nitin_test_plan",
  "version_description": null,
  "activated_environment": "none",
  "created_on": "2023-02-04T04:09:45.617",
  "created_by": "ngarg@mparticle.com",
  "last_modified_on": "2023-02-04T04:10:52.377",
  "last_modified_by": "ngarg@mparticle.com",
  "version_document": {
    "data_points": [
      {
        "description": "",
        "match": {
          "type": "custom_event",
          "criteria": {
            "event_name": "Media Event",
            "custom_event_type": "media"
          }
        },
        "validator": {
          "type": "json_schema",
          "definition": {
            "properties": {
              "data": {
                "additionalProperties": true,
                "properties": {},
                "required": [],
                "type": "object"
              }
            }
          }
        }
      },
      {
        "description": "",
        "match": {
          "type": "custom_event",
          "criteria": {
            "event_name": "Unknown Event",
            "custom_event_type": "unknown"
          }
        },
        "validator": {
          "type": "json_schema",
          "definition": {
            "properties": {
              "data": {
                "additionalProperties": true,
                "properties": {},
                "required": [],
                "type": "object"
              }
            }
          }
        }
      }
    ]
  }
}
```

## Master Issue
Closes https://go.mparticle.com/work/SQDSDKS-5064

### CODE REVIEW

File: test/general/testResults.ts [hunk 1]
```diff
@@ -555,10 +553,10 @@ mParticle.logEvent('CustomSocial', mParticle.EventType.Social)
 mParticle.logEvent('CustomOther', mParticle.EventType.Other)
 
 // Data Plan Point 10
-mParticle.logEvent('CustomUnknown', mParticle.EventType.Media)
+mParticle.logEvent('CustomUnknown', mParticle.EventType.Unknown)
 
 // Data Plan Point 11
-mParticle.logEvent('CustomMedia', mParticle.EventType.Other)
+mParticle.logEvent('CustomMedia', mParticle.EventType.Media)
```
Comment (rmi22186) [inline test/general/testResults.ts:559]: I think there should still be a test for `Other` since it is a valid event type

**Review by rmi22186 — CHANGES_REQUESTED** [review]
From a code perspective it looks fine with 1 minor comment. 

Also - minor nit, but I assumed that pulling down the demo app with everything would make it work, but after testing it and seeing the old behavior, I realized I had to do an `npm link` in the root, then `npm link package-name` in the snippets-demo.

File: test/general/testResults.ts (hunk 1 above)
Comment (alexs-mparticle) [reply test/general/testResults.ts:559]: I believe `Data Plan Point 9` tests for `Other` for each platform.

Comment (rmi22186) [reply test/general/testResults.ts:559]: ah, got it. didn't see that because it was hidden.  thanks for pointing that out!

**Review by rmi22186 — APPROVED** [review]

**Pull request merged**